include/zephyr/arch/riscv/structs.h

```diff
@@ -14,6 +14,9 @@ struct _cpu_arch {
 	unsigned long user_exc_tmp0;
 	unsigned long user_exc_tmp1;
 #endif
+#ifdef CONFIG_SMP
```

Comment:
why limit having access for hartid, just for SMP? for instance, if you wanted to to run a non smp zephyr application on a chosen hart in an AMP context for instance

Reply:
This is to know the hartid of another CPU. Only the SMP configuration would care about other CPUs. The UP and simple MP configurations are about Zephyr caring about its own CPU only and in that case `csr_read(mhartid)` to get its own is far more preferable.